File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        if (arr.size() <= 1) 
            return target == arr;
        
        sort(target.begin(),target.end());
        sort(arr.begin(), arr.end());
        
        for (int i = 0; i < arr.size(); i++) {
            if (target[i] != arr[i])
                return false;
            else
                continue;
        }
        
        return true;
    }
};

File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.rs
use std::collections::HashMap;

impl Solution {
    pub fn can_be_equal(target: Vec<i32>, arr: Vec<i32>) -> bool {
        let mut imap = HashMap::new();
        
        for i in 0..arr.len() {
            *imap.entry(target[i]).or_insert(0) += 1;
            *imap.entry(arr[i]).or_insert(0) -= 1;
        }
        
        for (num,freq) in imap {
            if freq != 0 {
                return false;
            }
        }
        
        true
    }
}